tee-worker/litentry/core/credentials/src/spaceid/mod.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use core::fmt::{Debug, Display};
+
+pub mod bnb_domain_holding_amount;
+
+pub fn match_domain_amount<T>(range: &[T], amount: T) -> Option<usize>
+where
+	T: std::cmp::PartialOrd + Display + Debug,
+{
+	let mut index = None;
+
+	for (idx, item) in range.iter().enumerate() {
```

Comment:
Can we use [position](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.position)?